### PR TITLE
[JENKINS-32504] JiraEnvironmentVariableBuilder and pipeline plugin

### PR DESCRIPTION
**Pull request by jan-zajic**
New pipeline step jiraIssueSelector and some documentation by examples.

[Review on Reviewable](https://reviewable.io/reviews/jenkinsci/jira-plugin/89)

### CODE REVIEW

**Comment by warden** [comment]
That's interesting workaround. 
@oleg-nenashev, @jglick any obvious review comments?
@jan-zajic any chance you can at least hit previous coverage?

File: COMPATIBILITY.md [hunk 1]
```diff
@@ -75,3 +97,4 @@ Other builders will be supported in future (not supported yet).
 - [ ] `JiraReleaseVersionUpdater` never supported
 - [ ] `JiraReleaseVersionUpdaterBuilder` not supported yet
 - [ ] `JiraVersionCreator` never supported
+- [ ] `JiraEnvironmentVariableBuilder` not supported
```
Comment (warden) [inline COMPATIBILITY.md]: + "(workaround)"

File: src/main/java/hudson/plugins/jira/pipeline/IssueSelectorStep.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+        @Override
+        public String getDisplayName() {
+            return Messages.IssueSelectorStep_Descriptor_DisplayName();
+        }
+    }
+
+    /**
+     * @author jan zajic
+     */
```
Comment (warden) [inline src/main/java/hudson/plugins/jira/pipeline/IssueSelectorStep.java]: no @author needed here

File: src/main/resources/hudson/plugins/jira/Messages.properties [hunk 3]
```diff
@@ -30,4 +30,5 @@ JiraEnvironmentVariableBuilder.DisplayName=JIRA: Add related environment variabl
 JiraEnvironmentVariableBuilder.NoJiraSite=[JIRA] No JIRA site is configured for this project. This must be a project configuration error
 JiraEnvironmentVariableBuilder.Updating=[JIRA] Setting {0} to {1}.
 CommentStep.Descriptor.DisplayName=JIRA: Add comment to the issue
-SearchIssuesStep.Descriptor.DisplayName=JIRA: Search issues
\ No newline at end of file
+SearchIssuesStep.Descriptor.DisplayName=JIRA: Search issues
+IssueSelectorStep.Descriptor.DisplayName=JIRA: Issue selector
```
Comment (warden) [inline src/main/resources/hudson/plugins/jira/Messages.properties]: newline

**Comment by oleg-nenashev** [comment]
sorry, missed the notification

**Pull request merged**